api/v1alpha1/orano2ims_types.go

```diff
@@ -23,6 +23,15 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type ResourceServer struct {
+	//+kubebuilder:default=false
+	Enabled bool `json:"enabled"`
+	//+optional
+	BackendURL string `json:"backendURL,omitempty"`
+	//+optional
+	BackendToken string `json:"backendToken,omitempty"`
+}
+
```

Comment:
Can we use an structure like this for all the other servers as well instead of the boolean that we use today? I'd suggest that we create a base structure containing only the `Enabled` boolean, and then we create structs specific for each server that embed it. For example:

```go
type ServerConfig struct {
        // Enabled indicates if the server should be started.
        //
        // +kubebuilder:default=true
        Enabled bool `json:"enabled"`
}

// ResourceServerConfig contains the configuration for the resource server.
type ResourceServerConfig struct {
        ServerConfig `json:",inline"`

        //+optional
	BackendURL string `json:"backendURL,omitempty"`
	//+optional
	BackendToken string `json:"backendToken,omitempty"`
}

// MetdataServerConfig contains the configuration for the metadata server.
type MetadataServerConfig struct {
        ServerConfig `json:",inline"`
}

// Same for all the other servers.
```

This will give us room to add other common parameters in the future. For example, at some point I would like to add logging settings:

```go
type ServerConfig struct {
        // Enabled indicates if the server should be started.
        //
        // +kubebuilder:default=true
        Enabled bool `json:"enabled"`

        // LogLevel indicates the log level.
        //
        // +kubebuilder:default=info
        LogLevel string `json:"logLevel"`
}
```

Reply:
Yes, that was my plan as well. Initially I was thinking to leave the common settings directly in spec, but we can have another structure if you prefer.

Reply:
Done.